File: flutter_app/lib/api/review_api.dart
import 'api_client.dart';

class ReviewApi {
  ReviewApi(this._client);

  final ApiClient _client;

  Future<void> addReview({
    required int swapRequestId,
    required int rating,
    String? comment,
  }) async {
    await _client.post('/review/add', body: {
      'swap_request_id': swapRequestId,
      'rating': rating,
      'comment': comment,
    });
  }

  Future<Map<String, dynamic>> userReviews(int userId) async {
    final data = await _client.get('/review/user/$userId') as Map<String, dynamic>;
    return data;
  }
}

File: frontend/src/api/reviewApi.ts
import { asRecord } from '../models/parse';
import { ApiClient } from './apiClient';

export class ReviewApi {
  constructor(private readonly client: ApiClient) {}

  async addReview(payload: {
    swapRequestId: number;
    rating: number;
    comment?: string;
  }): Promise<void> {
    await this.client.post('/review/add', {
      swap_request_id: payload.swapRequestId,
      rating: payload.rating,
      comment: payload.comment,
    });
  }

  async userReviews(userId: number): Promise<Record<string, unknown>> {
    return asRecord(await this.client.get(`/review/user/${userId}`));
  }
}
